registry/storage/cache/provider/cacheprovider.go

```diff
@@ -0,0 +1,39 @@
+package cacheprovider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/distribution/distribution/v3/registry/storage/cache"
+)
+
+// InitFunc is the type of a CacheProvider factory function and is
+// used to register the constructor for different CacheProvider backends.
+type InitFunc func(ctx context.Context, options map[string]interface{}) (cache.BlobDescriptorCacheProvider, error)
```

Comment:
Pet peeve: just call it what it is.
```suggestion
type Factory func(ctx context.Context, options map[string]interface{}) (cache.BlobDescriptorCacheProvider, error)
```

Reply:
this is called `InitFunc` because that's what we call similar funcs throughout the codebase e.g
* https://github.com/distribution/distribution/blob/04e3bdaa7c6304d14970f13c01f0a8ded8289e8d/registry/storage/driver/middleware/storagemiddleware.go#L12
* https://github.com/distribution/distribution/blob/04e3bdaa7c6304d14970f13c01f0a8ded8289e8d/registry/middleware/registry/middleware.go#L14

I'm not a major fan of it either but I think we should stay consistent

Reply:
Fair enough.